use-package.el

```diff
@@ -419,23 +419,23 @@ manually updated package."
   (let ((archive-symbol (if (symbolp archive) archive (intern archive)))
         (archive-name   (if (stringp archive) archive (symbol-name archive))))
     (if (use-package--archive-exists-p archive-symbol)
-        (push (cons package archive-name) package-pinned-packages)
+        (add-to-list 'package-pinned-packages (cons package archive-name) t)
```

Comment:
It should be safe. It has problems with local lexically bound variables, but there is no problem with globals. And `t` for `APPEND` wouldn't cause a problem, it just seems unneeded.

`unless member push` would be fine too, though a bit more verbose. `cl-pushnew` would be perfect, except it has the annoyance of requiring `cl-lib` at runtime (its expansion uses `cl-adjoin`).